langchain/base_language.py

```diff
@@ -51,6 +51,16 @@ async def agenerate_prompt(
     ) -> LLMResult:
         """Take in a list of prompt values and return an LLMResult."""
 
+    @abstractmethod
+    def predict(self, text: str, stop: Optional[List[str]] = None) -> str:
```

Comment:
Use * after positional args 
Change list to sequence

the default should be to get rid of all List types on all inputs